processors/src/main/java/io/opentelemetry/contrib/interceptor/common/Interceptable.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.interceptor.common;
+
+import io.opentelemetry.contrib.interceptor.api.Interceptor;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/** Base class to reuse the code related to intercepting signals. */
```

Comment:
I strongly believe that we should not use object inheritance as a mechanism for reusing code.